server/src/database.py

```diff
@@ -176,7 +177,7 @@ def pop_job(queue_list):
     """Get the next job in the queue"""
     # The queue name and the job are returned, but we don't need the queue now
     try:
-        response = mongo.db.jobs.find_one_and_update(
+        response = mongo.db.jobs.find_one(
```

Comment:
I was getting a weird issue in my test-case where the item that was returned in response was different from the item that was updated which was because I saw that update_one did not support that sort key, but on the MongoDB documentation, I see that the sort key functionality was added in v8.0(https://www.mongodb.com/docs/manual/reference/method/db.collection.findOneAndUpdate). Are we using v8.0 or does the mongomock also support this?

Reply:
I think this might be an issue with MongoMock:
https://github.com/mongomock/mongomock/blob/d821f320dbf91432a09a42dc200d7306ec5a6244/mongomock/collection.py#L1348

If you look at the implementation of find_one_and_update in MongoMock, there's two separate calls to find_one and update - the sort parameter is passed to the find_one function but not the update function.

Reply:
Actually not sure that this was the issue since the query changes, but I'll keep looking

Reply:
btw, I didn't see your question about which version earlier. Unfortunately, we are using mongo v5 (that's what was supported by the charm when we deployed it)